data/src/main/scala/io/prediction/data/webhooks/mailchimp/MailChimpConnector.scala

```diff
@@ -97,4 +107,200 @@ private[prediction] object MailChimpConnector extends FormConnector {
+      ("targetEntityId" -> data("data[list_id]")) ~
+      ("eventTime" -> eventTime) ~
+      ("properties" -> (
+        ("email" -> data("data[email]")) ~
+        ("email_type" -> data("data[email_type]")) ~
+        ("merges" -> (
+          ("EMAIL" -> data("data[merges][EMAIL]")) ~
+          ("FNAME" -> data("data[merges][FNAME]"))) ~
+          ("LNAME" -> data("data[merges][LNAME]")) ~
+          ("INTERESTS" -> data("data[merges][INTERESTS]"))
+        )) ~
+        ("ip_opt" -> data("data[ip_opt]")
+      ))
+
+    json
+
+  }
+
+  def upemailToEventJson(data: Map[String, String]): JObject = {
+
+    import org.json4s.JsonDSL._
+
+    /*
+    "type": "upemail", 
+    "fired_at": "2009-03-26 22:15:09", 
+    "data[list_id]": "a6b5da1054",
+    "data[new_id]": "51da8c3259", 
+    "data[new_email]": "api+new@mailchimp.com", 
+    "data[old_email]": "api+old@mailchimp.com"
+    */
+
+    // convert to ISO8601 format
+    val eventTime = Utils.dateTimeToString(parseMailChimpDateTime(data("fired_at")))
+
+    val json =
+      ("event" -> "upemail") ~
+      ("entityType" -> "user") ~
+      ("entityId" -> data("data[new_id]")) ~
+      ("targetEntityType" -> "list") ~
+      ("targetEntityId" -> data("data[list_id]")) ~
+      ("eventTime" -> eventTime) ~
+      ("properties" -> (
+        ("new_email" -> data("data[new_email]")) ~
+        ("old_email" -> data("data[old_email]"))
+      ))
+
+    json
+
+  }
+
+  def cleanedToEventJson(data: Map[String, String]): JObject = {
+
+    import org.json4s.JsonDSL._
+
+    /*
+    Reason will be one of "hard" (for hard bounces) or "abuse"
+    "type": "cleaned", 
+    "fired_at": "2009-03-26 22:01:00", 
+    "data[list_id]": "a6b5da1054",
+    "data[campaign_id]": "4fjk2ma9xd",
+    "data[reason]": "hard",
+    "data[email]": "api+cleaned@mailchimp.com"
+    */
+
+    // convert to ISO8601 format
+    val eventTime = Utils.dateTimeToString(parseMailChimpDateTime(data("fired_at")))
+
+    val json =
+      ("event" -> "cleaned") ~
+      ("entityType" -> "user") ~
+      ("entityId" -> data("data[campaign_id]")) ~
+      ("targetEntityType" -> "list") ~
+      ("targetEntityId" -> data("data[list_id]")) ~
+      ("eventTime" -> eventTime) ~
+      ("properties" -> (
+        ("reason" -> data("data[reason]")) ~
+        ("email" -> data("data[email]"))
+      ))
+
+    json
+
+  }
+
+  def campaignToEventJson(data: Map[String, String]): JObject = {
+
+    import org.json4s.JsonDSL._
+
+    /*
+    "type": "campaign", 
+    "fired_at": "2009-03-26 21:31:21", 
+    "data[id]": "5aa2102003",
+    "data[subject]": "Test Campaign Subject", 
+    "data[status]": "sent",
+    "data[reason]": "", 
+    "data[list_id]": "a6b5da1054"
+    */
+
+    // convert to ISO8601 format
+    val eventTime = Utils.dateTimeToString(parseMailChimpDateTime(data("fired_at")))
+
+    val json =
+      ("event" -> "campaign") ~
+      ("entityType" -> "user") ~
+      ("entityId" -> data("data[id]")) ~
```

Comment:
Thanks Grace. 
for "cleansed" event, How about also use "list" as entityType without targetEntityType?
ie. EntityType="list" has "cleaned" event with the email and campaign_id inside the properties.